hazelcast/src/main/java/com/hazelcast/query/impl/getters/JsonGetterContext.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl.getters;
+
+import com.hazelcast.json.internal.JsonPattern;
+import com.hazelcast.util.collection.WeightedEvictableList;
+import com.hazelcast.util.collection.WeightedEvictableList.WeightedItem;
+
+import java.util.List;
+
+public class JsonGetterContext {
+
+    private static final int PATTERN_CACHE_MAX_SIZE = 20;
+    private static final int PATTERN_CACHE_MAX_VOTES = 400;
+
+    private JsonPathCursor pathCursor;
+    private WeightedEvictableList<JsonPattern> patternList;
```

Comment:
I will change this to a ThreadLocal and get rid of complex/concurrent `WeightedEvictableList`. It brings about 70% performance improvement.